Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp

```diff
@@ -481,8 +481,9 @@ WebIDL::ExceptionOr<void> XMLHttpRequest::open(String const& method_string, Stri
     auto normalized_method = Fetch::Infrastructure::normalize_method(method);
 
     // 5. Let parsedURL be the result of parsing url with this’s relevant settings object’s API base URL and this’s relevant settings object’s API URL character encoding.
-    // FIXME: Pass in this’s relevant settings object’s API URL character encoding.
-    auto parsed_url = DOMURL::parse(url, HTML::relevant_settings_object(*this).api_base_url(), {});
+    auto api_base_url = HTML::current_settings_object().api_base_url();
+    auto api_url_character_encoding = HTML::current_settings_object().api_url_character_encoding();
+    auto parsed_url = DOMURL::parse(url, api_base_url, api_url_character_encoding);
```

Comment:
We should still be going this "this’s relevant settings object", rather than the "current settings object" here. The difference being that the relevant settings object is in the JS realm that this object was created, rather than whatever is the currently executing realm.

So something like:
```c++
auto& relevant_settings_object = HTML::relevant_settings_object(*this);
auto api_base_url = relevant_settings_object.api_base_url();
auto api_url_character_encoding = relevant_settings_object.api_url_character_encoding();
auto parsed_url = DOMURL::parse(url, api_base_url, api_url_character_encoding);
```

Reply:
You will also need to `clang-format` your code to appease the linter

Reply:
Thank you for the explanation regarding current vs relevant settings object, TIL. I've pushed another commit with the suggested fix, and also fixed the import order with `clang-format`.

Reply:
Almost there :) we also require that fixes in a PR be amended to the commit which is being fixed

From https://github.com/LadybirdBrowser/ladybird/blob/master/CONTRIBUTING.md:
> Amend your existing commits when adding changes after a review, where relevant.

The reason is to avoid having a bunch of "fixup" commits being merged into master. When we merge a PR, we like the end result of the PR to be a continuous set of working, atomic commits.

Reply:
Ah, sorry! I squashed the last commit. Thank you for your patience, explaining the process well to a first-time contributor is much appreciated :pray: